test/Microsoft.NET.Build.Tests/GivenThatWeWantToTargetNet471.cs

```diff
@@ -127,14 +159,120 @@ public void It_does_not_include_facades_from_nuget_packages()
+
+            buildCommand
+                .Execute("/v:normal")
+                .Should()
+                .Pass()
+                .And.NotHaveStdOutContaining("MSB3277") // MSB3277: Found conflicts between different versions of the same dependent assembly that could not be resolved.
+                .And.NotHaveStdOutContaining("Could not determine");
+
+            var outputDirectory = buildCommand.GetOutputDirectory(testProject.TargetFrameworks);
+
+            outputDirectory.Should().OnlyHaveFiles(new[] {
+                $"{testProject.Name}.exe",
+                $"{testProject.Name}.pdb",
+                $"{netStandardProject.Name}.dll",
+                $"{netStandardProject.Name}.pdb",
+                "System.Diagnostics.DiagnosticSource.dll" // This library will get pulled in as part of the closure of the ns16 project and will be copyied because it's not inbox.
+            }.Concat(net471Shims));
+        }
+
+        [Fact]
+        public void It_does_not_include_shims_when_app_references_471_library()
+        {
+            //  https://github.com/dotnet/sdk/issues/1625
+            if (!Net471ReferenceAssembliesAreInstalled())
+            {
+                return;
+            }
+            var testProject = new TestProject()
+            {
+                Name = "Net471App_Referencing_Net471Library",
+                TargetFrameworks = "net471",
+                IsSdkProject = true,
+                IsExe = true
+            };
+
+            var net471library = new TestProject()
+            {
+                Name = "Net471_Library",
+                TargetFrameworks = "net471",
+                IsSdkProject = true
+            };
+
+            testProject.ReferencedProjects.Add(net471library);
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject, "net471_ref_net471")
+                .Restore(Log, testProject.Name);
+
+            var buildCommand = new BuildCommand(Log, Path.Combine(testAsset.TestRoot, testProject.Name));
+
+            buildCommand
+                .Execute("/v:normal")
+                .Should()
+                .Pass()
+                .And.NotHaveStdOutContaining("MSB3277") // MSB3277: Found conflicts between different versions of the same dependent assembly that could not be resolved.
+                .And.NotHaveStdOutContaining("Could not determine");
+
+            var outputDirectory = buildCommand.GetOutputDirectory(testProject.TargetFrameworks);
+
+            outputDirectory.Should().OnlyHaveFiles(new[] {
+                $"{testProject.Name}.exe",
+                $"{testProject.Name}.pdb",
+                $"{net471library.Name}.dll",
+                $"{net471library.Name}.pdb",
+            });
+        }
+
+        static bool Net471ReferenceAssembliesAreInstalled()
+        {
+            var net461referenceAssemblies = ToolLocationHelper.GetPathToDotNetFrameworkReferenceAssemblies(TargetDotNetFrameworkVersion.Version461);
```

Comment:
I'd suggest adding a comment explaining this, ie: The version of the MSBuild libraries we are referencing doesn't have an enum value for .NET 4.7.1.  So we use the MSBuild API to find the path to the 4.6.1 reference assemblies, and locate the 4.7.1 reference assemblies relative to that.